meteor/client/ui/SegmentTimeline/SegmentTimelineContainer.tsx

```diff
@@ -375,13 +384,29 @@ export const SegmentTimelineContainer = translateWithTracker<IProps, IState, ITr
 				})
 				segment &&
 					this.subscribe(PubSub.pieces, {
-						startRundownId: segment.rundownId,
-						startSegmentId: { $in: Array.from(this.props.segmentsIdsBefore.values()) },
 						invalid: {
 							$ne: true,
 						},
-						// same rundown, and previous segment
-						lifespan: { $in: [PieceLifespan.OutOnRundownEnd, PieceLifespan.OutOnRundownChange] },
+						$or: [
+							// same rundown, and previous segment
+							{
+								startSegmentId: { $in: Array.from(this.props.segmentsIdsBefore.values()) },
```

Comment:
I think this part of the query could do with a startRundownId condition too, to help scope it down